keps/sig-autoscaling/2021-scale-from-zero/README.md

```diff
@@ -143,138 +181,142 @@ updates.
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
 
-This issue has been raised multiple times ([#116984][], [#125987][], [#62013][],
-[#aks-3068][], [#keda-1100][]), with users commenting that:
+- Provide scaling to zero replicas for object and external metrics
+- Provide scaling from zero replicas for object and external metrics
 
-1. For large deployments, a 10% tolerance translates into very significant resources (i.e. hundreds of pods).
-2. This tolerance can slow down scaling operations, hindering responsiveness in
-case of surges.
-3. Scale-ups are more a problem than scale-downs since typically pods are slower to initialize than to shut down, and since responding to load increase is typically more critical than freeing resources.
+### Non-Goals
 
-Since appropriate tolerance values are workload-dependent, this KEP proposes to let users add custom tolerance values to `HorizontalPodAutoscaler` resources, overriding the existing default value when present.
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
 
-This solution integrates seamlessly with the existing HPA API since it already allows users to [fine-tune the autoscaler behavior](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#configurable-scaling-behavior).
-The exact API recommended here has been previously proposed in [kep-853][] (see [here](https://github.com/kubernetes/enhancements/pull/1234#discussion_r333036990)), but it was then decided to implement it separately.
+- Provide scaling to/from zero replicas for resource metrics
+- Provide request buffering at the Kubernetes Service level
 
-[#116984]: https://github.com/kubernetes/kubernetes/issues/116984
-[#125987]: https://github.com/kubernetes/kubernetes/issues/125987
-[#62013]: https://github.com/kubernetes/kubernetes/issues/62013
-[#aks-3068]: https://github.com/Azure/AKS/issues/3068
-[#keda-1100]: https://github.com/kedacore/keda-docs/issues/1100
-[Kube Control Manager]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
-[kep-853]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/853-configurable-hpa-scale-velocity/README.md
+## Proposal
 
-### Goals
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
 
-- Allow users to optionally override the default workload autoscaling tolerance on a per-HPA bases.
+Allow the HPA to scale from and to zero using `minReplicas: 0` and a HPA status condition.
```

Comment:
a little more detail would be helpful

Reply:
Could you elaborate whether you mean more detail why this is needed or how this is implemented?

The mechanism is currently described below, but I can move it here as well.